packages/core/src/types.ts

```diff
@@ -119,6 +119,7 @@ export type ComponentRegistration = {
 
 export type ComponentRegistrationOptions = {
   enabledBuiltInComponents?: string[];
+  experimentalComponents?: Record<string, boolean>;
```

Comment:
I wonder if it would be better to define the object explicitly since it only has 1 expected property 🤔 
```suggestion
  experimentalComponents?: {
    carousel?: boolean;
  };
```